lib/syskit/models/deployment_group.rb

```diff
@@ -225,12 +225,18 @@ def register_configured_deployment(configured_deployment)
 
             # Enumerates all the deployments registered on self
             #
+            # @param [String] on the process server names to look for. If empty, all
+            #   process server names will be considered.
+            # @param [String] except_on the process server names to be ignored.
             # @yieldparam [ConfiguredDeployment]
-            def each_configured_deployment
-                return enum_for(__method__) unless block_given?
+            def each_configured_deployment(on: [], except_on: [])
+                return enum_for(__method__, on: on, except_on: except_on) unless block_given?
 
                 deployments.each_value do |set|
-                    set.each { |c| yield(c) }
+                    set.each do |c|
+                        yield(c) if !on.empty? && on.include?(c.process_server_name)
+                        yield(c) if on.empty? && !except_on.include?(c.process_server_name)
```

Comment:
This hides that the two conditions are mutually exclusive ... which is weird. Maybe

```
if on.empty?
   yield(c) if except_on.include?(c.process_server_name)
else
   yield(c) if on.include?(c.process_server_name)
end
```